index.js

```diff
@@ -328,6 +328,21 @@ Parser.prototype._parse = function (line) {
   if (line === '\n')
     return
 
+  // assert from previous line
+  if (this.current) {
+    var indent = line.match(/^[ \t]+/)
+    if (indent) {
+      indent = indent[0]
+      // not beginning of yamlish, and not already in yamlish
+      if (line !== indent + '---\n' && !this.yind && !this.yamlish) {
+        this.emitResult()
```

Comment:
Fails to parse valid tap.

```
TAP version 13
1..1
not ok 1 - please keep my diags
  # the diags are coming
  ---
  after: comment
  ...
```

This will cause it to lose the diags and emit them as "extra" lines.